src/WebJobs.Extensions.MobileApps/Config/MobileAppsConfiguration.cs

```diff
@@ -53,8 +62,194 @@ public void Initialize(ExtensionConfigContext context)
                 throw new ArgumentNullException("context");
             }
 
-            context.Config.RegisterBindingExtension(
-                new MobileTableAttributeBindingProvider(context.Config, this, context.Config.NameResolver));
+            INameResolver nameResolver = context.Config.GetService<INameResolver>();
+            IConverterManager converterManager = context.Config.GetService<IConverterManager>();
+
+            BindingFactory factory = new BindingFactory(nameResolver, converterManager);
+
+            IBindingProvider outputProvider = factory.BindToGenericAsyncCollector<MobileTableAttribute, MobileTableContext>(typeof(MobileTableAsyncCollector<>), BindForOutput);
+
+            IBindingProvider clientProvider = factory.BindToExactType<MobileTableAttribute, IMobileServiceClient>(BindForClient);
+
+            IBindingProvider queryProvider = factory.BindToGenericItem<MobileTableAttribute>(BindForQuery);
+            queryProvider = factory.AddTypeFilter(IsQueryType, queryProvider);
+            queryProvider = factory.AddValidator<MobileTableAttribute>(ThrowIfGenericArgumentIsInvalid, queryProvider);
```

Comment:
Suggestions for Validators/TypeFilters --

1. Can we have a single 'Filter' concept here that passes me all the context I'll need to make a decision. For now, that context is the (%% resolved) Attribute and the parameter Type.

The sig could be Func<TAttribute, Type, bool> and the contract:
- Return true if this matches. Call next filter in the line.
- Return false if this attribute does not match and do not continue. Move onto the next rule.
- Throw if you want to abort indexing of this parameter completely. For example: The type is correct but you haven't set an 'Id' property on the attribute. You don't want any other rule to claim this parameter so you throw.

If all return true, then this rule has been matched.
If any return false, move onto the next rule.

2. I'd expect the Filters to be called before the binding is created -- I think right now the TypeFilter is called before creation and Validators are called after?

3. Since these are wrapping existing BindingProviders, I believe the Validators are called in reverse-order (last one added is the first one called). It would feel more natural if we could build up a list of Filters/Validators and have them called in the order they were added.